### DISCUSSION TITLE
Applications "pinned to start menu" do not appear in start menu

### DISCUSSION THREAD

**Discussion by demdante** [valinet/ExplorerPatcher#1085] [General]
Hello all.  First, I want to thank the developer of this software for opening up the discussions for the repo after, I assume, reading the email I sent.  Really appreciate it.  So, the bug that I wanted to report involves attempting to add applications to the start menu via the "pin to start" context menu option when you right click an application in the start menu **or** search menu.  Specifically, applications which you search for using the search button are not added to the start menu after selecting the "pin to start" context menu option for said application.  To better demonstrate the problem, the below graphic demonstrates how the process works in vanilla (not running explorer patcher) windows 11:

![VAnilla Windows 11](https://media.giphy.com/media/fz5CB8TMXMkZ9RONkK/giphy.gif)

However, this is what happens when you try to do the exact same thing when on a computer using the latest version of explorer patcher  using the explorer patcher start menu:

![Explorer Patcher](https://media.giphy.com/media/Qb6HVnFILXwvli3Xxm/giphy.gif)

As you can see, the application is added to the start menu in the vanilla windows 11, but not when windows 11 is using explorer patcher.  So, you might think "well, why not just add the application from the start menu if you are having issues pinning the application from the search menu."  Well, unfortunately, for some reason, the windows start menu does not work like it used to in all the previous versions of windows.  For some reason, in windows 11, there are folders/applications located in C:\ProgramData\Microsoft\Windows\Start Menu\Programs that, for whatever reason, do not appear in the windows 11 start menu.  An example of one such folder (and all of the apps contained within that folder) is C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Windows Accessories, which contains applications like the Remote Desktop application, meaning that, as explorer patcher exists now, we can't pin the remote desktop app to the start menu because it does not exist in the start menu's list of applications and can only be found by searching for it.  I have also attached the reg file (converted to a text doc) containing the explorer patcher settings that I had configured when running the test demonstrated above for your convenience.  Let me know if you need me to provide/do anything else.  Thanks.
[ExplorerPatcher_22622.450.50.2.reg.txt](https://github.com/valinet/ExplorerPatcher/files/9580041/ExplorerPatcher_22622.450.50.2.reg.txt)

**Comment by valinet** [comment]
It’s explained in a couple of places, I don’t remember where exactly; the data stores for the 2 menus are different. The library that links the interaction to the data store from Windows 11 only knows how to talk to the Windows 11 menu. To have it recognize the Windows 10 menu, an older version of the library should be used. This would be the default in ExplorerPatcher, but unfortunately the old version has some side effects. Thus, EP only provides a workaround to this: open EP Properties - Restart File Explorer, then go to the Start menu section- select Windows 10 style - Pin tiles to Start menu. A File Explorer window opens; use that to browse to the item you want added, right click and Pin to Start from there will work and will pin to the Windows 10 menu. Try twice it doesn’t appear first try.

Discussions on GitHub are actually closed, please follow up on email if you need further assistance.

Best regards,

**Reply by Erosion139** [reply]
This does not work with everything. I am trying to pin blender to the start menu but even through file explorer it will not pin. However, a folder will gladly pin itself to the start menu. I don't know what the difference is, both should be pinnable.

**Reply by kuriandungu** [reply]
I've had the same issue for ages. this video helped me fix it. https://youtu.be/guoaKWX3mj8?si=X2y7ppINs5OakpFi 
in summary I ran these 3 commands below in CMD in Admin mode. it clears all your current Quick access data and you have to re-pin them.  After running the commands Re-Pinning started working. (I didnt even need to reboot pc)

del /F /Q %APPDATA%\Microsoft\Windows\Recent\*

del /F /Q %APPDATA%\Microsoft\Windows\Recent\AutomaticDestinations\*

del /F /Q %APPDATA%\Microsoft\Windows\Recent\CustomDestinations\

**Reply by Erosion139** [reply]
> I've had the same issue for ages. this video helped me fix it. https://youtu.be/guoaKWX3mj8?si=X2y7ppINs5OakpFi in summary I ran these 3 commands below in CMD in Admin mode. it clears all your current Quick access data and you have to re-pin them. After running the commands Re-Pinning started working. (I didnt even need to reboot pc)
> 
> del /F /Q %APPDATA%\Microsoft\Windows\Recent*
> 
> del /F /Q %APPDATA%\Microsoft\Windows\Recent\AutomaticDestinations*
> 
> del /F /Q %APPDATA%\Microsoft\Windows\Recent\CustomDestinations\

This would be a great fix if it had anything to do with the post...

**Reply by pyrates999** [reply]
%APPDATA%\Microsoft\Windows\Recent\AutomaticDestinations is where all the jump lists are stored for each application you have installed.

Do not run those commands.

**Comment by mattdoot** [comment]
ran the 3 cmds n the issue still persists

**Reply by GladTech** [reply]
Hello, not sure if you figured it out or moved on, but I just posted an easy solution below, so check it out if you're interested!

**Comment by GladTech** [comment]
If anyone is still looking for a simple method to add or removed pinned programs to the start menu, I just found a [website](https://www.neowin.net/news/explorerpatcher-sets-windows-10-start-menu-right-but-breaks-windows-11s/) that easily explains it. 

Essentially, you open your Win 10 start menu, and then drag the program over to the pinned area to pin it.
To unpin it, drag the program down on the pinned area to a new group, which should be a highlighted section. Then at the top of the group where you can add a name, just right-click and select unpin group.

It's really that simple and I don't know if it's new or not, but can't believe that no one has this info readily available for something as simple as pinning an app. 

Other than that though, the Explorer Patcher is an amazing program that I'm absolutely loving!
Thank you devs and everyone involved!

**Reply by dulapeepss** [reply]
Just to document a clarification for anyone struggling as I was, this doesn’t work in full screen mode, so you’ll have to switch to “default” / “start menu” mode to make any changes.

**Comment by GladTech** [comment]
> Hello all. First, I want to thank the developer of this software for opening up the discussions for the repo after, I assume, reading the email I sent. Really appreciate it. So, the bug that I wanted to report involves attempting to add applications to the start menu via the "pin to start" context menu option when you right click an application in the start menu **or** search menu. Specifically, applications which you search for using the search button are not added to the start menu after selecting the "pin to start" context menu option for said application. To better demonstrate the problem, the below graphic demonstrates how the process works in vanilla (not running explorer patcher) windows 11:
> 
> ![VAnilla Windows 11](https://camo.githubusercontent.com/cccf409751214a72bc5f55b17c3c8911b0fb763315174252c673e8093f7f7032/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f667a35434238544d584d6b5a39524f4e6b4b2f67697068792e676966) [ ![VAnilla Windows 11](https://camo.githubusercontent.com/cccf409751214a72bc5f55b17c3c8911b0fb763315174252c673e8093f7f7032/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f667a35434238544d584d6b5a39524f4e6b4b2f67697068792e676966) ](https://camo.githubusercontent.com/cccf409751214a72bc5f55b17c3c8911b0fb763315174252c673e8093f7f7032/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f667a35434238544d584d6b5a39524f4e6b4b2f67697068792e676966) [ ](https://camo.githubusercontent.com/cccf409751214a72bc5f55b17c3c8911b0fb763315174252c673e8093f7f7032/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f667a35434238544d584d6b5a39524f4e6b4b2f67697068792e676966)
> 
> However, this is what happens when you try to do the exact same thing when on a computer using the latest version of explorer patcher using the explorer patcher start menu:
> 
> ![Explorer Patcher](https://camo.githubusercontent.com/58989ea2450af8c4a57141792887ed188552935b7a361887796c4db7cbfcc029/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f51623648566e46494c5877766c693358786d2f67697068792e676966) [ ![Explorer Patcher](https://camo.githubusercontent.com/58989ea2450af8c4a57141792887ed188552935b7a361887796c4db7cbfcc029/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f51623648566e46494c5877766c693358786d2f67697068792e676966) ](https://camo.githubusercontent.com/58989ea2450af8c4a57141792887ed188552935b7a361887796c4db7cbfcc029/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f51623648566e46494c5877766c693358786d2f67697068792e676966) [ ](https://camo.githubusercontent.com/58989ea2450af8c4a57141792887ed188552935b7a361887796c4db7cbfcc029/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f51623648566e46494c5877766c693358786d2f67697068792e676966)
> 
> As you can see, the application is added to the start menu in the vanilla windows 11, but not when windows 11 is using explorer patcher. So, you might think "well, why not just add the application from the start menu if you are having issues pinning the application from the search menu." Well, unfortunately, for some reason, the windows start menu does not work like it used to in all the previous versions of windows. For some reason, in windows 11, there are folders/applications located in C:\ProgramData\Microsoft\Windows\Start Menu\Programs that, for whatever reason, do not appear in the windows 11 start menu. An example of one such folder (and all of the apps contained within that folder) is C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Windows Accessories, which contains applications like the Remote Desktop application, meaning that, as explorer patcher exists now, we can't pin the remote desktop app to the start menu because it does not exist in the start menu's list of applications and can only be found by searching for it. I have also attached the reg file (converted to a text doc) containing the explorer patcher settings that I had configured when running the test demonstrated above for your convenience. Let me know if you need me to provide/do anything else. Thanks. [ExplorerPatcher_22622.450.50.2.reg.txt](https://github.com/valinet/ExplorerPatcher/files/9580041/ExplorerPatcher_22622.450.50.2.reg.txt)

Hello, not sure if you figured it out or moved on, but I just posted an easy solution below, so check it out if you're interested!
